src/Database/Schema/TableSchema.php

```diff
@@ -773,6 +778,55 @@ public function getConstraint(string $name): ?array
         return $this->_constraints[$name] ?? null;
     }
 
+    /**
+     * Get a constraint object for a given constraint name.
+     *
+     * Constraints have a few subtypes such as foreign keys and primary keys.
+     * You can either use `instanceof` or getType() to check for subclass types.
+     *
+     * @param string $name The name of the constraint to get.
+     * @return \Cake\Database\Schema\Constraint A constraint object.
+     */
+    public function constraint(string $name): Constraint
+    {
+        $data = $this->getConstraint($name);
+        if ($data === null) {
+            $message = sprintf(
+                'Table `%s` does not contain a constraint named `%s`.',
+                $this->_table,
+                $name,
+            );
+            throw new DatabaseException($message);
+        }
+        $data['name'] = $data['constraint'] ?? $name;
+        unset($data['constraint']);
```

Comment:
I'd like to preserve the name of primary keys if it is available. I still need to think through how to handle backwards compatibility and in which way to do the aliasing.